vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -260,4 +261,16 @@ public void setIndeterminate(boolean indeterminate) {
     public boolean isIndeterminate() {
         return isIndeterminateBoolean();
     }
+
+    /**
+     * Replaces the label content with the given label component.
+     *
+     * @param component
+     *            the component to be added to the label.
+     */
+    public void setLabelComponent(Component component) {
+        getElement().appendChild(labelElement.getElement());
```

Comment:
We should call `setLabel("")` here, same as we do in `setLabelAsHtml()`. Otherwise after `setLabel("foo"); setLabelComponent(new Text("bar"))`, the `getLabel()` would still return "foo".

Reply:
So `getLabel()` method couldn't be used when using label component?

Reply:
Yes, IMO it makes sense to align it with how `setLabelAsHtml()` works.